src/signature.cpp

```diff
@@ -32,8 +32,16 @@ std::vector<uint8_t> SignatureUtils::create(AsymmetricPrivateKey &privateKey,
 {
     std::vector<uint8_t> messageDigest;
 
-    if(privateKey.getType() != AsymmetricKey::KeyTypes::RSA){
-        throw MoCOCrWException("Functionality is only supported for RSA keys");
+    if(privateKey.getType() != AsymmetricKey::KeyTypes::RSA) {
```

Comment:
I fear all of this was refactored in master and needs to merged here as well before we can do this :(